### ISSUE TITLE
Shaders Begin "On" But Are Actually Off Until Retoggled 

### ISSUE THREAD

**Issue by dottiekh** [IrisShaders/Iris#1244]
### What happened?

When first loading into a world after startup of Minecraft, Iris does not load the selected shader pack until it is toggled off and on again, at which point they resume working. This behavior isn't consistent, as sometimes shaders begin on as they should.

### Relevant log output

```shell
[Render thread/ERROR]: Couldn't load NewWorldRenderingPipeline, falling back to vanilla shaders.
java.lang.NullPointerException: at index 0
	at com.google.common.collect.ObjectArrays.checkElementNotNull(ObjectArrays.java:229) ~[guava-31.0.1-jre.jar:?]
	at com.google.common.collect.ObjectArrays.checkElementsNotNull(ObjectArrays.java:219) ~[guava-31.0.1-jre.jar:?]
	at com.google.common.collect.ObjectArrays.checkElementsNotNull(ObjectArrays.java:213) ~[guava-31.0.1-jre.jar:?]
	at com.google.common.collect.ImmutableList.construct(ImmutableList.java:353) ~[guava-31.0.1-jre.jar:?]
	at com.google.common.collect.ImmutableList.copyOf(ImmutableList.java:265) ~[guava-31.0.1-jre.jar:?]
	at net.coderbot.iris.gl.program.ProgramUniforms$Builder.buildUniforms(ProgramUniforms.java:266) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.ExtendedShader.<init>(ExtendedShader.java:45) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewShaderTests.create(NewShaderTests.java:119) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.createShader(NewWorldRenderingPipeline.java:357) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.createShader(NewWorldRenderingPipeline.java:347) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.lambda$new$11(NewWorldRenderingPipeline.java:280) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.ShaderMap.<init>(ShaderMap.java:21) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.newshader.NewWorldRenderingPipeline.<init>(NewWorldRenderingPipeline.java:271) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.Iris.createPipeline(Iris.java:557) ~[iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.coderbot.iris.pipeline.PipelineManager.preparePipeline(PipelineManager.java:34) [iris-mc1.18.1-1.2.0-pre.jar:?]
	at net.minecraft.class_761.handler$bba000$iris$beginLevelRender(class_761.java:6453) [client-intermediary.jar:?]
	at net.minecraft.class_761.method_22710(class_761.java:1240) [client-intermediary.jar:?]
	at net.minecraft.class_757.method_3188(class_757.java:1031) [client-intermediary.jar:?]
	at net.minecraft.class_757.method_3192(class_757.java:811) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1117) [client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:733) [client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:238) [client-intermediary.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:608) [fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:77) [fabric-loader-0.12.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.12.12.jar:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:?]
	at jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[?:?]
	at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:?]
	at java.lang.reflect.Method.invoke(Method.java:568) ~[?:?]
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:210) [NewLaunch.jar:?]
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:245) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.listen(EntryPoint.java:143) [NewLaunch.jar:?]
	at org.multimc.EntryPoint.main(EntryPoint.java:34) [NewLaunch.jar:?]
```

### Minecraft Version

Minecraft 1.18.1

### Iris Version

iris-mc1.18.1-1.2.0-pre.jar

### Sodium Version

sodium-fabric-mc1.18.1-0.4.0-alpha6+build.14.jar

### Operating System

Windows 10

### What is your GPU?

Nvidia GeForce RTX 2060

### Java Version

Adoptium OpenJDK Java 17.0.1.12-hotspot

### Additional context

Other mods enabled:
```
AppleSkin 1.18-2.2.0
Architectuary 3.4.9
CITResewn
Continuity 1.0.31.18
Dynamic FPS 2.1.0
Effective 1.1
FerriteCore 4.0.0
Figura 0.0.8-rc.2-1.18
Firstperson Fabric 2.1.0
Indium 1.0.2-alpha2
Light Overlay 6.0.4
Mambience 3.2.0
Modmenu 3.0
MouseWheelie 3.2.0
NotEnoughAnimations 1.8.0
Ok Zoomer 5.0.0-beta.3
Right Click Harvest 1.5.1
Roughly Enough Items 7.1.390
Server Pack Unlocker
Starlight 1.0.0
Xaero's Minimap 21.22.3.1
Xaero's World Map 1.18.6.1
```

**Comment by DioEgizio** [comment]
What shaderpack are you using?

**Comment by dottiekh** [comment]
Complementary Shaders v4.3.3

**Comment by coderbot16** [comment]
This should have been resolved in https://github.com/IrisShaders/Iris/commit/216f13f22aed84f3d7242b57b138ccdb4571add8, which was released in Iris 1.2.3 (the latest version is Iris 1.2.4).
